tools/integration_tests/rapid_appends/reads_after_appends_test.go

```diff
@@ -17,17 +17,66 @@ package rapid_appends
 import (
 	"fmt"
 	"log"
+	"math/rand/v2"
 	"os"
 	"path"
 	"syscall"
+	"testing"
 	"time"
 
 	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/operations"
 	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/setup"
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
+////////////////////////////////////////////////////////////////////////
+// Helpers
+////////////////////////////////////////////////////////////////////////
+
+// declare a function type for read and verify
+type readAndVerifyFunc func(t *testing.T, filePath string, expectedContent []byte)
+
+func readSequentiallyAndVerify(t *testing.T, filePath string, expectedContent []byte) {
+	readContent, err := operations.ReadFileSequentially(filePath, 1024*1024)
+
+	// For sequential reads, we expect the content to be exactly as expected.
+	require.NoErrorf(t, err, "failed to read file %q sequentially: %v", filePath, err)
+	require.Equal(t, readContent, expectedContent)
+}
+
+func readRandomlyAndVerify(t *testing.T, filePath string, expectedContent []byte) {
+	file, err := operations.OpenFileAsReadonly(filePath)
+	require.NoErrorf(t, err, "failed to open file %q: %v", filePath, err)
+	defer operations.CloseFileShouldNotThrowError(t, file)
+	if len(expectedContent) == 0 {
+		t.SkipNow()
+	}
+	fileInfo, err := file.Stat()
+	require.NoError(t, err)
+	fileSize := fileInfo.Size()
+	require.GreaterOrEqualf(t, fileSize, int64(len(expectedContent)), "file %q is too small to read %d bytes", filePath, len(expectedContent))
+
+	// Ensure offset and readSize are within bounds of expected content to be read.
+	maxOffset := int(len(expectedContent))
+	// Limit number of reads if the file is too small.
+	numReads := min(maxOffset, 10)
+	for i := range numReads {
+		offset := rand.IntN(maxOffset)
+		readSize := rand.IntN(int(fileSize - int64(offset)))
```

Comment:
shouldn't you use len(expectedcontent) instead of filesize. what will happen when offset=0